R/r-lib.R

```diff
@@ -0,0 +1,49 @@
+#' Helpers from the r-lib organisation
```

Comment:
I propose we call this file `lifecycle.R` and reword the docs to talk about the "lifecycle terminology used by the tidyverse and r-lib packages".  This way we explain where it comes from but others are welcome to use it too.

I think this link should appear somewhere:
https://www.tidyverse.org/lifecycle/